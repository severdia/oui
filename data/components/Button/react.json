{"description":"","displayName":"Button","methods":[],"props":{"ariaLabel":{"type":{"name":"string"},"required":false,"description":"Describes buttons that have an icon but no text"},"children":{"type":{"name":"node"},"required":true,"description":"Text within the button"},"isActive":{"type":{"name":"bool"},"required":false,"description":"Render button with active state"},"isDisabled":{"type":{"name":"bool"},"required":false,"description":"Prevent users from interacting with the button","defaultValue":{"value":"false","computed":false}},"isSubmit":{"type":{"name":"bool"},"required":false,"description":"Make the button act as a submit button","defaultValue":{"value":"false","computed":false}},"onClick":{"type":{"name":"func"},"required":false,"description":"Function that fires when the button is clicked on"},"size":{"type":{"name":"enum","value":[{"value":"'tiny'","computed":false},{"value":"'small'","computed":false},{"value":"'large'","computed":false},{"value":"'narrow'","computed":false},{"value":"'tight'","computed":false}]},"required":false,"description":"Various height and width options"},"style":{"type":{"name":"enum","value":[{"value":"'highlight'","computed":false},{"value":"'danger'","computed":false},{"value":"'danger-outline'","computed":false},{"value":"'outline'","computed":false},{"value":"'outline-reverse'","computed":false},{"value":"'plain'","computed":false},{"value":"'toggle'","computed":false},{"value":"'underline'","computed":false},{"value":"'unstyled'","computed":false}]},"required":false,"description":"Various color options"},"testSection":{"type":{"name":"string"},"required":false,"description":"Hook for automated JavaScript tests"},"width":{"type":{"name":"enum","value":[{"value":"'default'","computed":false},{"value":"'full'","computed":false}]},"required":false,"description":"Various height and width options","defaultValue":{"value":"'default'","computed":false}}},"private":false}