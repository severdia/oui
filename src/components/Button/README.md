## Button

Buttons are the main component people use to take action in our products. For example, they let people create experiments, remove goals, and close dialogs.

![alt text](http://design.optimizely.com/assets/img/design-patterns/product/buttons/button-example-creation.png "Logo Title Text 1")


## Button Styles
Our standard button style is a rectangular button that includes text, an icon, or both.
