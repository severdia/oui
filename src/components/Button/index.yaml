name: Button
description: Buttons are the main component people use to take action in our products. For example, they let people create experiments, remove goals, and close dialogs.
