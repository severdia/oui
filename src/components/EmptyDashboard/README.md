# EmptyDashboard Component

## Empty states
Users encounter empty states the first time they use our products or a new feature. It’s our opportunity to educate and delight our customers, as well as alleviate their fears and anxieties.

A good empty state will inform people what this page is, why they should care, and what their next step should be. Linking to educational content, such as Knowledge Base articles and blog posts, is a great way to make our customers successful. Finally, a good empty state should also have an illustrated graphic to visually explain what the feature does, and save the screen from being just text.

A poor empty state, on the other hand, is one that is just blank, or simply tells the user there is no data.

## Good Example

![alt text](http://design.optimizely.com/assets/img/design-patterns/product/empty-states/example-good.png "empty state example")

The empty Campaigns screen welcomes people to Optimizely Personalization with an illustration, briefly explains what a campaign is, and provides a call-to-action to create a campaign.

![alt text] (http://design.optimizely.com/assets/img/design-patterns/product/empty-states/example-sidebar.png "empty state example")

Sidebars should also have an empty state that explains what the feature does, what the user's next step is, and an illustration.

## Bad Example 

![alt text] (http://design.optimizely.com/assets/img/design-patterns/product/empty-states/example-bad.png "bad example")

The empty state of the Uploaded Lists feature is poor for many reasons. First, it simply states that the user has 0 lists and is using 0 bytes of space. This is dry and not helpful. Second, it states, “No uploaded lists match your filters. Try clearing them to see all of your uploaded lists.” This statement is incorrect and confusing because the user isn’t filtering anything — there simply aren't any lists uploaded yet. The sidebar tells people to select a list, but there aren’t any lists to select. Finally, there’s a missed opportunity to educate users about what this feature is, how to use it, and why they should use it.

## Oui React *EmptyDashboard* Component