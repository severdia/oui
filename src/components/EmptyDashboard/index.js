import React from 'react';
import PropTypes from 'prop-types';

const EmptyDashboard = ({
  button,
  imagePath,
  headline,
  description,
  descriptionMaxWidth,
  testSection,
}) => {
  return (
    <div
      data-oui-component={ true }
      className='flex overflow-y--scroll flex--1'
      data-test-section='layers-empty-state'
      style={{ width: '750px', margin: 'auto' }}>
      <div className='anchor--middle height--1-1 flex flex--column'>

        <div
          data-oui-component={ true }
          className='text--center soft-quad'
          data-test-section={ testSection }>

          { imagePath && (
            <div
              className="muted push-double--bottom"
              style={{ width: '550px', margin: 'auto' }}>
              <object
                data={ imagePath }
                className="svg--non-scaling-stroke display--inline"
                style={{ maxWidth: '450px', maxHeight: '290px' }}
                data-test-section={ testSection && `${testSection}-image` }
                alt=""
              />
            </div>
          ) }

          { headline && (
            <h1
              className="alpha push--bottom"
              data-test-section={ testSection && `${testSection}-headline` }>
              { headline }
            </h1>
          ) }

          { description && (
            <div
              className="muted push--bottom anchor--middle"
              style={{ maxWidth: descriptionMaxWidth }}
              data-test-section={ testSection && `${testSection}-description` }>
              { description }
            </div>
          ) }

          { button && (
            <div className="push-double--top">
              { button }
            </div>
          ) }
        </div>

      </div>
    </div>
  );
};

EmptyDashboard.propTypes = {
  /** A button, link, or other element that is actionable */
  button: PropTypes.node,
  /** A string or `div` containing longer text that describes the empty state */
  description: PropTypes.element,
  /** Max width of description. Accepts any valid CSS width value. */
  descriptionMaxWidth: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  /** Short text about the empty state */
  headline: PropTypes.string.isRequired,
  /** Path to an image representing the empty state */
  imagePath: PropTypes.string,
  /** Identifier used to create data-test-section attributes for testing */
  testSection: PropTypes.string,
};

EmptyDashboard.getDefaultProps = {
  descriptionMaxWidth: '450px',
  testSection: 'empty-hello',
};

export default EmptyDashboard;
