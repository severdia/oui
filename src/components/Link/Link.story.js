import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';

import Link from './index.js';
import Button from '../Button';

const stories = storiesOf('Link', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories
  .add('Default', (() => {
    return (
      <ul>
        <li>
          <Link
            href={ text('href', 'http://google.com') }
            isDisabled={ boolean('isDisabled', false) }
            style={ select('style',
              {
                default: 'default',
                dark: 'dark',
                muted: 'muted',
                'bad-news': 'bad-news',
                reverse: 'reverse',
              }, 'default') }>
            { text('children', 'Link') }
          </Link>
        </li>
      </ul>
    );
  }))
  .add('New window', (() => {
    return (
      <ul>
        <li>
          <Link
            href={ text('href', 'http://google.com') }
            newWindow={ boolean('newWindow', true) }
            style={ select('style',
              {
                default: 'default',
                dark: 'dark',
                muted: 'muted',
                'bad-news': 'bad-news',
                reverse: 'reverse',
              }, 'default') }>
            { text('children', 'Link') }
          </Link>
        </li>
      </ul>
    );
  }))
  .add('All links', () => {
    return (
      <ul>
        <li><Link href="http://google.com" isDisabled={ boolean('isDisabled', false) } onClick={ action('click blocklist item home') }>{ text('default', 'Link') }</Link></li>
        <li><Link href="http://google.com" isDisabled={ boolean('isDisabled', false) } style="dark">{ text('dark', 'Link Dark') }</Link></li>
        <li><Link href="http://google.com" isDisabled={ boolean('isDisabled', false) } style="muted">{ text('muted', 'Link Muted') }</Link></li>
        <li><Link href="http://google.com" isDisabled={ boolean('isDisabled', false) } style="bad-news">{ text('bad-news', 'Link Bad News') }</Link></li>
        <li className="background--brand"><Link href="http://google.com" isDisabled={ boolean('isDisabled', false) } style="reverse">{ text('reverse', 'Link Reverse') }</Link></li>
      </ul>
    );
  })
  .add('All disabled', () => {
    return (
      <ul>
        <li><Link href="http://google.com" isDisabled={ true }>Link</Link></li>
        <li><Link href="http://google.com" isDisabled={ true } style="dark">Link Dark</Link></li>
        <li><Link href="http://google.com" isDisabled={ true } style="muted">Link Muted</Link></li>
        <li><Link href="http://google.com" isDisabled={ true } style="bad-news">Link Bad News</Link></li>
        <li className="background--brand"><Link href="http://google.com" isDisabled={ true } style="reverse">Link Reverse</Link></li>
      </ul>
    );
  })
  .add('Link that looks like a Button', () => {
    return (
      <Link
        newWindow={ true }
        href={ text('href', 'http://google.com') }>
        <Button onClick={ null }>Go To Google</Button>
      </Link>
    );
  });

storiesOf('Link (Fake)', module)
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ))
  .add('Enabled', () => {
    return (
      <Link>Faux Link</Link>
    );
  })
  .add('Disabled', () => {
    return (
      <Link
        isDisabled={ true }>
        Faux Link
      </Link>
    );
  });
