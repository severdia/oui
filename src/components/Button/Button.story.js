import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, select, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import ButtonRow from '../ButtonRow';
import Button from './index.js';
import Link from '../Link';
import ArrowsInline from '../ArrowsInline';

const stories = storiesOf('Button', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories
  .add('Knobs', (() => {
    return (
      <Button
        style={ select('style', {
          'highlight': 'highlight',
          'danger': 'danger',
          'danger-outline': 'danger-outline',
          'outline': 'outline',
          'outline-reverse': 'outline-reverse',
          'plain': 'plain',
          'toggle': 'toggle',
          'underline': 'underline',
          'unstyled': 'unstyled',
        }, 'highlight') }
        isDisabled={ boolean('isDisabled', false) }
        isActive={ boolean('isActive', false) }
        isLoading={ boolean('isLoading', false) }
        width={ select('width', {
          'default': 'default',
          'full': 'full'}, 'default') }
        size={ select('size', {
          'tiny': 'tiny',
          'small': 'small',
          'large': 'large',
          'narrow': 'narrow',
          'tight': 'tight'}, 'narrow') }>
        { text('customize me!', 'customize me!') }
      </Button>);
  }))
  .add('All buttons', (() => {
    return (
      <ButtonRow
        centerGroup={ [
          <Button key="1" isActive={ boolean('isActive', false) } width="default">Button</Button>,
          <Button key="2" isActive={ boolean('isActive', false) } style="highlight" width="default">Highlight</Button>,
          <Button key="3" isActive={ boolean('isActive', false) } style="danger" width="default">Danger</Button>,
          <Button key="4" isActive={ boolean('isActive', false) } style="danger-outline" width="default">Danger Outline</Button>,
          <Button key="5" isActive={ boolean('isActive', false) } style="outline" width="default">Outline</Button>,
          <Button key="6" isActive={ boolean('isActive', false) } style="plain" width="default">Plain</Button>,
          <Button key="7" isActive={ boolean('isActive', false) } style="toggle" width="default">Toggle</Button>,
        ] }
      />);
  }))
  .add('Disabled', () => {
    return (
      <ButtonRow
        centerGroup={ [
          <Button key="1" isDisabled={ true } width="default">Button</Button>,
          <Button key="2" isDisabled={ true } style="highlight" width="default">Highlight</Button>,
          <Button key="3" isDisabled={ true } style="danger" width="default">Danger</Button>,
          <Button key="4" isDisabled={ true } style="danger-outline" width="default">Danger Outline</Button>,
          <Button key="5" isDisabled={ true } style="outline" width="default">Outline</Button>,
          <Button key="6" isDisabled={ true } style="plain" width="default">Plain</Button>,
          <Button key="7" isDisabled={ true } style="toggle" width="default">Toggle</Button>,
        ] }
      />);
  })
  .add('Underline button', () => {
    return (
      <Button style="underline" width="default">
        Underline Button
        <ArrowsInline direction="down" />
      </Button>);
  })
  .add('Sizes variations', () => {
    return (
      <ButtonRow
        centerGroup={ [
          <Button key="1" width="default" size="tiny">Tiny Button</Button>,
          <Button key="2" width="default" size="small">Small Button</Button>,
          <Button key="3" width="default" size="large">Large Button</Button>,
          <Button key="4" width="default" size="narrow">Narrow Button</Button>,
          <Button key="5" width="default" size="tight">Tight Button</Button>,
        ] }
      />);
  })
  .add('Button without an onClick', () => {
    return (
      <Button isLink={ true }>Faux Button</Button>
    );
  })
  .add('Button without an onClick (disabled)', () => {
    return (
      <Button isLink={ true } isDisabled={ true }>Faux Button</Button>
    );
  })
  .add('Button that looks like a Link', (() => {
    return (
      <Button
        style='unstyled'
        onClick={ action('Saying hi!') }>
        <Link>Say Hi</Link>
      </Button>
    );
  }))
  .add('Button that disables on click', () => {
    return (
      <Button
        style="highlight"
        isLoading={ boolean('isLoading', true) }
        onClick={ action('I have been clicked') }>
          Create Campaign
      </Button>
    );
  })
  .add('Button that disables on click with custom text', () => {
    return (
      <Button
        style="highlight"
        isLoading={ boolean('isLoading', true) }
        loadingText={ text('loadingText', 'Creating Campaign') }
        onClick={ action('I have been clicked') }>
          Create Campaign
      </Button>
    );
  });
