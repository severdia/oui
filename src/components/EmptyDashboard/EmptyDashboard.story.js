import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, text, select } from '@storybook/addon-knobs';

import Button from '../Button/';
import Link from '../Link/';
import EmptyDashboard from './index.js';

const stories = storiesOf('Empty Dashboard', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories
  .add('Default empty', () => {
    return (
      <EmptyDashboard
        headline={ text('headline', 'Get started with Optimizely X') }
        description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
        button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
        imagePath={ select('imagePath', {
          'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg': 'Campaigns',
          'https://app.optimizely.com/static/img/x/empty-states/extensions-empty-state.svg': 'Extensions',
          'https://app.optimizely.com/static/img/x/empty-states/custom-project-experiments-empty-state.svg': 'SST',
          'https://app.optimizely.com/static/img/x/empty-states/empty-rollouts.svg': 'Rollouts',
          'https://app.optimizely.com/static/img/x/empty-states/empty-feature-flags.svg': 'Feature Flags',
          'https://app.optimizely.com/static/img/x/empty-states/audience-list-empty-state.svg': 'Audiences',
          'https://app.optimizely.com/static/img/x/empty-states/attribute-dashboard-empty-state.svg': 'Attribute',
          'https://app.optimizely.com/static/img/x/empty-states/page-list-empty-state.svg': 'Page List',
        }, 'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg') }
      />
    );
  })
  .add('All images', () => {
    return (
      <div>
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
          button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ select('imagePath', {
            'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg': 'Campaigns',
            'https://app.optimizely.com/static/img/x/empty-states/extensions-empty-state.svg': 'Extensions',
            'https://app.optimizely.com/static/img/x/empty-states/custom-project-experiments-empty-state.svg': 'SST',
            'https://app.optimizely.com/static/img/x/empty-states/empty-rollouts.svg': 'Rollouts',
            'https://app.optimizely.com/static/img/x/empty-states/empty-feature-flags.svg': 'Feature Flags',
            'https://app.optimizely.com/static/img/x/empty-states/audience-list-empty-state.svg': 'Audiences',
            'https://app.optimizely.com/static/img/x/empty-states/attribute-dashboard-empty-state.svg': 'Attribute',
            'https://app.optimizely.com/static/img/x/empty-states/page-list-empty-state.svg': 'Page List',
          }, 'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg') }
        />
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
          button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ select('imagePath', {
            'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg': 'Campaigns',
            'https://app.optimizely.com/static/img/x/empty-states/extensions-empty-state.svg': 'Extensions',
            'https://app.optimizely.com/static/img/x/empty-states/custom-project-experiments-empty-state.svg': 'SST',
            'https://app.optimizely.com/static/img/x/empty-states/empty-rollouts.svg': 'Rollouts',
            'https://app.optimizely.com/static/img/x/empty-states/empty-feature-flags.svg': 'Feature Flags',
            'https://app.optimizely.com/static/img/x/empty-states/audience-list-empty-state.svg': 'Audiences',
            'https://app.optimizely.com/static/img/x/empty-states/attribute-dashboard-empty-state.svg': 'Attribute',
            'https://app.optimizely.com/static/img/x/empty-states/page-list-empty-state.svg': 'Page List',
          }, 'https://app.optimizely.com/static/img/x/empty-states/extensions-empty-state.svg') }
        />
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
          button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ select('imagePath', {
            'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg': 'Campaigns',
            'https://app.optimizely.com/static/img/x/empty-states/extensions-empty-state.svg': 'Extensions',
            'https://app.optimizely.com/static/img/x/empty-states/custom-project-experiments-empty-state.svg': 'SST',
            'https://app.optimizely.com/static/img/x/empty-states/empty-rollouts.svg': 'Rollouts',
            'https://app.optimizely.com/static/img/x/empty-states/empty-feature-flags.svg': 'Feature Flags',
            'https://app.optimizely.com/static/img/x/empty-states/audience-list-empty-state.svg': 'Audiences',
            'https://app.optimizely.com/static/img/x/empty-states/attribute-dashboard-empty-state.svg': 'Attribute',
            'https://app.optimizely.com/static/img/x/empty-states/page-list-empty-state.svg': 'Page List',
          }, 'https://app.optimizely.com/static/img/x/empty-states/custom-project-experiments-empty-state.svg') }
        />
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
          button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ select('imagePath', {
            'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg': 'Campaigns',
            'https://app.optimizely.com/static/img/x/empty-states/extensions-empty-state.svg': 'Extensions',
            'https://app.optimizely.com/static/img/x/empty-states/custom-project-experiments-empty-state.svg': 'SST',
            'https://app.optimizely.com/static/img/x/empty-states/empty-rollouts.svg': 'Rollouts',
            'https://app.optimizely.com/static/img/x/empty-states/empty-feature-flags.svg': 'Feature Flags',
            'https://app.optimizely.com/static/img/x/empty-states/audience-list-empty-state.svg': 'Audiences',
            'https://app.optimizely.com/static/img/x/empty-states/attribute-dashboard-empty-state.svg': 'Attribute',
            'https://app.optimizely.com/static/img/x/empty-states/page-list-empty-state.svg': 'Page List',
          }, 'https://app.optimizely.com/static/img/x/empty-states/empty-rollouts.svg') }
        />
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ select('imagePath', {
            'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg': 'Campaigns',
            'https://app.optimizely.com/static/img/x/empty-states/extensions-empty-state.svg': 'Extensions',
            'https://app.optimizely.com/static/img/x/empty-states/custom-project-experiments-empty-state.svg': 'SST',
            'https://app.optimizely.com/static/img/x/empty-states/empty-rollouts.svg': 'Rollouts',
            'https://app.optimizely.com/static/img/x/empty-states/empty-feature-flags.svg': 'Feature Flags',
            'https://app.optimizely.com/static/img/x/empty-states/audience-list-empty-state.svg': 'Audiences',
            'https://app.optimizely.com/static/img/x/empty-states/attribute-dashboard-empty-state.svg': 'Attribute',
            'https://app.optimizely.com/static/img/x/empty-states/page-list-empty-state.svg': 'Page List',
          }, 'https://app.optimizely.com/static/img/x/empty-states/empty-feature-flags.svg') }
        />
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ select('imagePath', {
            'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg': 'Campaigns',
            'https://app.optimizely.com/static/img/x/empty-states/extensions-empty-state.svg': 'Extensions',
            'https://app.optimizely.com/static/img/x/empty-states/custom-project-experiments-empty-state.svg': 'SST',
            'https://app.optimizely.com/static/img/x/empty-states/empty-rollouts.svg': 'Rollouts',
            'https://app.optimizely.com/static/img/x/empty-states/empty-feature-flags.svg': 'Feature Flags',
            'https://app.optimizely.com/static/img/x/empty-states/audience-list-empty-state.svg': 'Audiences',
            'https://app.optimizely.com/static/img/x/empty-states/attribute-dashboard-empty-state.svg': 'Attribute',
            'https://app.optimizely.com/static/img/x/empty-states/page-list-empty-state.svg': 'Page List',
          }, 'https://app.optimizely.com/static/img/x/empty-states/audience-list-empty-state.svg') }
        />
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ select('imagePath', {
            'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg': 'Campaigns',
            'https://app.optimizely.com/static/img/x/empty-states/extensions-empty-state.svg': 'Extensions',
            'https://app.optimizely.com/static/img/x/empty-states/custom-project-experiments-empty-state.svg': 'SST',
            'https://app.optimizely.com/static/img/x/empty-states/empty-rollouts.svg': 'Rollouts',
            'https://app.optimizely.com/static/img/x/empty-states/empty-feature-flags.svg': 'Feature Flags',
            'https://app.optimizely.com/static/img/x/empty-states/audience-list-empty-state.svg': 'Audiences',
            'https://app.optimizely.com/static/img/x/empty-states/attribute-dashboard-empty-state.svg': 'Attribute',
            'https://app.optimizely.com/static/img/x/empty-states/page-list-empty-state.svg': 'Page List',
          }, 'https://app.optimizely.com/static/img/x/empty-states/attribute-dashboard-empty-state.svg') }
        />
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ select('imagePath', {
            'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg': 'Campaigns',
            'https://app.optimizely.com/static/img/x/empty-states/extensions-empty-state.svg': 'Extensions',
            'https://app.optimizely.com/static/img/x/empty-states/custom-project-experiments-empty-state.svg': 'SST',
            'https://app.optimizely.com/static/img/x/empty-states/empty-rollouts.svg': 'Rollouts',
            'https://app.optimizely.com/static/img/x/empty-states/empty-feature-flags.svg': 'Feature Flags',
            'https://app.optimizely.com/static/img/x/empty-states/audience-list-empty-state.svg': 'Audiences',
            'https://app.optimizely.com/static/img/x/empty-states/attribute-dashboard-empty-state.svg': 'Attribute',
            'https://app.optimizely.com/static/img/x/empty-states/page-list-empty-state.svg': 'Page List',
          }, 'https://app.optimizely.com/static/img/x/empty-states/page-list-empty-state.svg') }
        />
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath='https://app.optimizely.com/static/img/x/empty-states/change-history-empty-state.svg'
        />
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath='https://app.optimizely.com/static/img/x/empty-states/campaign-manager-empty.svg'
        />
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath='https://app.optimizely.com/static/img/x/empty-states/custom-snippets-empty-state.svg'
        />
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath='https://app.optimizely.com/static/img/x/empty-states/fullstack-variables-empty-state.svg'
        />
        <EmptyDashboard
          headline={ text('headline', 'Get started with Optimizely X') }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath='https://app.optimizely.com/static/img/x/empty-states/integrations-empty.svg'
        />
      </div>
    );
  });
