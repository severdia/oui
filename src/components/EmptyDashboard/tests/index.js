import React from 'react';
import Button from '../../Button/';
import EmptyDashboard from '../index.js';
import Link from '../../Link/';
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('components/EmptyDashboard', () => {
  it('should render', () => {
    const output = mount(
      <EmptyDashboard
        headline='Get started with Optimizely X'
        description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">Highlight Button</Button>) }
        imagePath='https://app.optimizely.com/static/img/p13n/campaign-list-empty-state.svg'
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

});
