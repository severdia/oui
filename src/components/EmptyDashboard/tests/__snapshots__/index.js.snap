// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/EmptyDashboard should render 1`] = `
<EmptyDashboard
  button={
    <Button
      isDisabled={false}
      isLink={false}
      isLoading={false}
      isSubmit={false}
      loadingText=""
      onBlur={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      style="highlight"
      width="default"
    >
      Highlight Button
    </Button>
  }
  description={
    <div>
      This is a test description about 
      <Link
        href="http://google.com"
      >
        X Feature
      </Link>
       with a Google.
    </div>
  }
  headline="Get started with Optimizely X"
  imagePath="https://app.optimizely.com/static/img/p13n/campaign-list-empty-state.svg"
>
  <div
    className="flex overflow-y--scroll flex--1"
    data-oui-component={true}
    data-test-section="layers-empty-state"
    style={
      Object {
        "margin": "auto",
        "width": "750px",
      }
    }
  >
    <div
      className="anchor--middle height--1-1 flex flex--column"
    >
      <div
        className="text--center soft-quad"
        data-oui-component={true}
      >
        <div
          className="muted push-double--bottom"
          style={
            Object {
              "margin": "auto",
              "width": "550px",
            }
          }
        >
          <object
            alt=""
            className="svg--non-scaling-stroke display--inline"
            data="https://app.optimizely.com/static/img/p13n/campaign-list-empty-state.svg"
            style={
              Object {
                "maxHeight": "290px",
                "maxWidth": "450px",
              }
            }
          />
        </div>
        <h1
          className="alpha push--bottom"
        >
          Get started with Optimizely X
        </h1>
        <div
          className="muted push--bottom anchor--middle"
          style={
            Object {
              "maxWidth": undefined,
            }
          }
        >
          <div>
            This is a test description about 
            <Link
              href="http://google.com"
            >
              <a
                className="link"
                data-oui-component={true}
                disabled={false}
                href="http://google.com"
              >
                X Feature
              </a>
            </Link>
             with a Google.
          </div>
        </div>
        <div
          className="push-double--top"
        >
          <Button
            isDisabled={false}
            isLink={false}
            isLoading={false}
            isSubmit={false}
            loadingText=""
            onBlur={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
            style="highlight"
            width="default"
          >
            <button
              aria-live="polite"
              className="oui-button oui-button--highlight oui-button--default"
              data-oui-component={true}
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
              type="button"
            >
              Highlight Button
            </button>
          </Button>
        </div>
      </div>
    </div>
  </div>
</EmptyDashboard>
`;
